### DISCUSSION TITLE
How to check the RS-232 pins on the controller

### DISCUSSION THREAD

**Discussion by Spuds** [jgyates/genmon#1117] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

1.19.02

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

The basic info:
- I've had `GenMon` working for many years, its awesome.
- Running a `RPI zero W` with a `max3232`.  
- My cable gets its power from the battery and the signal from the `RS232`  7,8 pins on the controller.
- It also uses the free (read useless) `Generac mobile link` as this generator came with a WiFi module (its 4 wires are still wired, I think those are 1,2,5,6)

Recently we had a near by lighting strike, as in very close. It took out an LG TV, (2) UniFi switches, StarLink controller, Rachio sprinkler controller, GFI outlet, TV antenna booster,  UniFi access point, Carrier HVAC and an RPI running piHole.   There is a whole house surge protector plus spot surge protectors on all electronics.  Fortunately all of those are fine LOL

Now onto the `GenMon` ... its not communicating, every attempt is s a timeout error.
- I checked the cable and did find a broken inline coupler (I connect the `Generac` cable to the RPI with RG6 cables/connectors, the coupler is just in place to allow for easy swap out). ... No improvement
- I changed the `max3232` with a new one (its on a socket on a Hat so easy) No improvement.
- I had a new `RPI Zero 2W`, so I installed a fresh copy of bullseye and then current GenMon, checked that `serialtest.py` passes.  No improvement.

The free mobile link does indicate a battery voltage, outdoor temperature, self maintenance test time ... I'm assuming that is successfully communicating with the controller via RS485. 

Finally the questions!
- Is there a way to test the `RS232` pins on the controller to see if they are working (with a good DVM)
- Same question but for the `RS485` ?
- IF the `RS232` pins are fried from the lighting strike AND the `RS485` are still working, is there a way to have `GenMon` use the `RS485` pins (such as using a 232<->485 conversion box or other ?)
- I hope I'm not looking at having to replace the controller as its otherwise functioning fine (it did its job during a outage, menus all respond, etc) all because a $0.20 chip popped on the `RS232` side.
- Other things I should check

Thanks in advance for help and guidance.

**Comment by jgyates** [comment]
@Spuds 

The RS-485 port communicates via a different protocol so genmon will not work on that port.

Some max3232 boards have LEDs for TX and RX. If this not available you could look with a multimeter on the RX pin on the controller. It should be between -3V and -15V when idle and fluctuate between the -3V / -15V at the low point to +3V / +12V at the hight point. Typical is around +/- 6V.

RS-458 / RS-422 is differential so you are looking at two wires for TX and two wire for RX. For RS- 422, TX is the difference between TX+ and TX- pins. They are the inverted signal of each other and range between 0 and 5V. Rs485 the output will be floating (tri-state) when idle so you will not get a valid signal until something is transmitted, unless there are bias resistors used. Then you would read between 0V and 5V. The voltage levels are the same for RS-485.

**Comment by Spuds** [comment]
I back probed the 8 pin `Molex` connector on the controller, using the `-` probe on the battery neg post.  The voltages were steady (Fluke 87V)
```
Pin 01: 5.3 VDC Steady
Pin 02: 0  VDC Steady
Pin 03: 13 VDC Steady
Pin 04: 0 VDC Steady
Pin 05: 2.4 VDC Steady
Pin 06: 2.5 VDC Steady
Pin 07: 0 VDC Steady
Pin 08: 6.5 VDC Steady
```

I took the same measurements (back probe at the controller) with the factory WiFi plugged in ( pins 1,2,5,6) minimal changes
```
Pin 01: 5 VDC Steady
Pin 02: 0  VDC Steady
Pin 03: 13 VDC Steady
Pin 04: 0 VDC Steady
Pin 05: 1.9 VDC Steady
Pin 06: 2 VDC Steady
Pin 07: 0 VDC Steady
Pin 08: 6.6  VDC Steady
```

Just for completeness, probing at the RPI, again no fluctuation.
```
Tx 6.45
Rx 0.106
POS 12.88 
NEG 0.02
```

So thoughts ... seems like the RS-232 is gone, and maybe the RS-485 as well?

**Reply by skipfire** [reply]
I had a ground strike that did similar damage in my home (network gear, cameras, garage door openers) and it fried the controller on the generator.  From the display on the controller it seemed ok, but when running it didn't regulate very well and neither comms path worked.
